backend/gamify/views.py

```diff
@@ -1,3 +1,65 @@
+from django.core import serializers
+from django.http import HttpResponse, JsonResponse
 from django.shortcuts import render
+from rest_framework.decorators import api_view
+from rest_framework.parsers import JSONParser
+from rest_framework.viewsets import ModelViewSet
 
-# Create your views here.
+from gamify.models import Team, User
+from gamify.serializers import UserSerializer
+
+
+@api_view(['POST'])
+def createTeam(request):
+    """
+    Create a team
+    inputs: JSON with two field "name", "user_id"
+    """
+    data = JSONParser().parse(request)
+    new_team = Team.objects.create(name=data['name'])
+    new_team.save()
+
+    user = User.objects.get(id=data["user_id"])
+    user.team = new_team
+    user.team_leader = True
+    user.save()
+
+    return JsonResponse(serializers.serialize("json", new_team), status=201)
+ 
+
+@api_view(['POST'])
+def leaveTeam(request):
+    """
+    Leave a team
+    inputs: JSON with one field "user_id"
+    """
+    data = JSONParser().parse(request)
+    user = User.objects.get(id=data["user_id"])
+    # TODO: Hacky, easier to deal with data consistency by not letting team leaders leave
+    if user.team_leader:
+        return JsonResponse(serializers.serialize("json", user), status=403)
+    user.team = None
```

Comment:
do we wanna set team_leader as None or leave it as false? just food for thought.